File: src/my_project/medium_problems/from1to50/jump_game_ii.py
from typing import List

class Solution:
    def jump(self, nums: List[int]) -> int:

        target = len(nums) - 1
        count = 0

        while target > 0:
            idx = 0

            while nums[idx] + idx < target:
                idx += 1

            target = idx

            count += 1

        return count


solution = Solution()

print(solution.jump([1,1,1,1,1,1,1,10,1,1]))

File: my_project_typescript/medium_problems_typescript/from1to50/jump_game_ii.ts
function jump(nums: number[]): number {


    let target: number = nums.length -1 
    let count: number = 0
    let idx: number 
    while (target > 0){
        idx = 0

        while (nums[idx] + idx < target){
            idx += 1
        }
        target = idx
        count += 1
    }
    return count 
};